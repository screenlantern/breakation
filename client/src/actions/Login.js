import axios from 'axios';

export function login(credentials){
    axios.post( '/api/users/login' , credentials)
    .then((response) => {
        return {
            type: 'LOGGED_IN',
            payload: response
        };
    })
    .catch((response)=> {
        return {
            type: 'LOGIN_FAILED',
            payload: response
        };
    });
}