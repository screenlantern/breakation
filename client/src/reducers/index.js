import { combineReducers } from 'redux';
import sessionReducer from './rd_session';

const rootReducer = combineReducers({
  login:  sessionReducer
});

export default rootReducer;
