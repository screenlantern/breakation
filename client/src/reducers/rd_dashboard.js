export const InitialState = {
    username: '',
    admin: false
};

export default function(state=InitialState, action){
    switch(action.type){
        case REGISTER_SUCCESS:
            return {...state, success: true};
        case REGISTER_FAIL:
            return {...state, success: false};
        default:
            return state;
    }
}