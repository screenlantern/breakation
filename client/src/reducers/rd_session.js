import { AUTH_USER,  
         UNAUTH_USER,
         AUTH_ERROR,
         PROTECTED_TEST } from '../actions/types';

export const InitialState = {
    loggedIn: false,
    shouldRedirect: false,
    errorMessage: null
}

export default function(state = InitialState, action){
    switch(action.type){
        case AUTH_USER:
            return Object.assign({}, state, { loggedIn: true, shouldRedirect: true, data: action.payload });
        case UNAUTH_USER:
            return { ...state,  loggedIn: false, shouldRedirect: false, errorMessage: 'authorization failed'};
        default:
            return state;
    }

}

