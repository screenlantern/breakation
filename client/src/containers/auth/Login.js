import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import { StyleSheet, css } from 'aphrodite';
import Button  from '../../components/button/Button';
import {login} from '../../actions/ac_session';


class Login extends Component {
    constructor(props) {
        super(props);

        this.state = {
            credentials: {
                email: '',
                password: ''
            }
        };

        this.onInputChange = this.onInputChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

    }

    componentWillReceiveProps(nextProps) {
        //if (nextProps.status.loggedIn){ this.props.history.push({pathname: '/dashboard' }) };
    }


    onInputChange(e) {
        const creds = this.state.credentials;
        creds[e.target.name] = e.target.value;
        this.setState({
            credentials: creds
        });
    }

    onSubmit(e) {
        e.preventDefault();
        console.log(this.props);
        this.props.login(this.state.credentials, this.props.history);
    }

    render() {
        return (
            <div>
                <form className={css(styles.form)} onSubmit={this.onSubmit} onChange={this.onInputChange}>
                    <div className={css(styles.div)}>
                        <label>Email</label>
                        <input className={css(styles.input)} type="email" placeholder="Type your email address here" name="email" value={this.state.credentials.email}
                               required/>
                    </div>
                    <div className={css(styles.div)}>
                        <label>Password</label>
                        <input className={css(styles.input)} type="password" placeholder="Type password here" name="password"
                               value={this.state.credentials.password} required/>
                    </div>
                    <p className={css(styles.p, styles.right)}>Forgot your password?<a>&nbsp;&nbsp;Remind</a></p>
                    <Button type="submit" color="primary">Log in</Button>
                    <p className={css(styles.p)}>Don't have an account?<a href="/registration">&nbsp;&nbsp;Sign up</a></p>
                </form>
            </div>
        );
    }
}

const styles = StyleSheet.create({
    form: {
        maxWidth: '404px',
        maxHeight: '472px',
        margin: '0 auto',
        transform: 'translateY(50%)',
        textAlign: 'center',
        fontFamily: '\'Lato\', sans-serif;',
        color: '#31383D',
        overflow: 'hidden'
    },
    input: {
        height: '48px',
        width: '100%',
        border: '1px solid #D5DBE2',
        fontSize: '18px',
        color: 'rgba(151,164,177,0.5)',
        fontWeight: '300',
        padding: '0.5rem',
        boxSizing: 'border-box'
    },
    div: {
        textAlign: 'left'
    },
    right: {
      textAlign: 'right'
    },
    p: {
        fontSize: '21px',
        color: '#B1BECD',
        width: '100%',
        ':first-of-Type':{
            marginTop: '0',
            marginBottom: '60px'
        },
        ':last-of-Type': {
            marginTop: '60px'
        }
    }
});

const mapDispatchToProps = function (dispatch, ownProps) {
    return bindActionCreators({login}, dispatch);
};

const mapStateToProps = function (state, ownProps) {
    return {status: state.session};
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
