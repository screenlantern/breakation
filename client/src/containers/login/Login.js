import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { login } from '../../actions/Login';

class Login extends Component {
  constructor(props){
    super(props);
  }

  render(){
    return (
      <div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return bindActionCreators({ login }, dispatch);
}

export default connect(null, mapDispatchToProps)(Login);
