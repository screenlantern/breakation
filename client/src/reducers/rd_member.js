import { ADD_SUCCESS, ADD_FAIL } from '../actions/types';

export const InitialState = {
    success: false
};

export default function(state=InitialState, action){
    switch(action.type){
        case ADD_SUCCESS:
            return {...state, success: true};
        case ADD_FAIL:
            return {...state, success: false};
        default:
            return state;
    }
}