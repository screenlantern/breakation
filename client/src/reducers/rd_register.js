import { REGISTER_SUCCESS, REGISTER_FAIL } from '../actions/types';

export const InitialState = {
    success: false
};

export default function(state=InitialState, action){
    switch(action.type){
        case REGISTER_SUCCESS:
            return {...state, success: true};
        case REGISTER_FAIL:
            return {...state, success: false};
        default:
            return state;
    }
}