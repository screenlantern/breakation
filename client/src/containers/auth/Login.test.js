import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Login from './Login';
import {initialState as sessionSate } from '../../reducers/rd_session';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


it('render with default props', () => {
    const store = mockStore({ sessionSate });
    const wrapper = shallow(<Login store={store} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should dispatch an action to authenticate.js user', () => {

});

