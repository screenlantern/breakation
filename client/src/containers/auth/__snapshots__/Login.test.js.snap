// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with default props 1`] = `
<Login
  login={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
